base/reduce.jl

```diff
@@ -762,6 +762,73 @@ Inf
 """
 minimum(a; kw...) = mapreduce(identity, min, a; kw...)
 
+"""
+    extrema(itr; [init]) -> Tuple
+
+Compute both the minimum and maximum element in a single pass, and return them as a 2-tuple.
+
+The value returned for empty `itr` can be specified by `init`. It must be a 2-tuple whose
+first and second elements are neutral elements for `min` and `max` respectively
+(i.e. which are greater/less than or equal to any other element). It is used for non-empty
+collections. Note: it implies that, for empty `itr`, the first element is typically
+_greater_ than the last element. This is a "paradoxical" but yet expected result.
```

Comment:
Not sure this is ideal, but what about something like this?

```suggestion
The value returned for empty `itr` can be specified by `init`. It must be a 2-tuple whose
first and second elements are neutral elements for `min` and `max` respectively
(i.e. which are greater/less than or equal to any other element). As a consequence, when `itr` is empty the
returned `(mn, mx)` tuple will satisfy `mn > mx`. When `init` is specified it is used even for non-empty `itr`.
```

Reply:
Hmm... We'd need something like `reduce_empty(::typeof(max), ::Type{T}) where {T} = typemin(T)` for "when `itr` is empty the returned `(mn, mx)` tuple will satisfy `mn > mx`" to make sense, right?  (I commented in https://github.com/JuliaLang/julia/issues/36266#issuecomment-644424299 but I'm in favor of _not_ defining this `reduce_empty`.)

But I think naming the returned value is a good idea! So I stole it and updated the docstrings: 360b3463cef1f03681ffdbffbb6cb005f9ac09f4

Reply:
I don't think it has anything to do with `reduce_empty`? By supplying `init` you bypass it, right?

Maybe it's easier if I explain the problem. Having "It is used for non-empty collections" come between your specification of the neutral elements and the consequences for the returned value feels like an interruption of thought. I recognize these issues are interleaved (the reason you need certain sentinel values is because they are used for non-empty collections), but this version of the docstring seems unnecessarily confusing.

Reply:
Thanks, after re-reading your version again, I think it makes sense now. It's added to the patch 3ceaaae94bb5a26c36a72c5e824f90c1821ca8a4